ms2query/ms2library.py

```diff
@@ -0,0 +1,375 @@
+            set and as value the value this attribute should have.
+
+        """
+        # Get all attributes to check if the arguments in settings are allowed
+        allowed_arguments = self.__dict__
+        # Set all kwargs as attributes, when in allowed_arguments
+        for key in settings:
+            assert key in allowed_arguments, \
+                f"Invalid argument in constructor:{key}"
+            assert isinstance(settings[key], type(allowed_arguments[key])), \
+                f"Different type is expected for argument: {key}"
+            setattr(self, key, settings[key])
+
+    def pre_select_spectra(self,
+                           query_spectra: List[Spectrum]):
+        """Returns dict of dataframe with preselected spectra
+
+        The structure of the panda dataframe is
+
+        Args:
+        ------
+        query_spectra:
+            spectra for which a preselection of possible library matches should
+            be done
+        """
+
+        # spec2vec_similarities_scores = self._get_spec2vec_similarity_matrix(
+        #     query_spectra)
+        same_masses = self._get_parent_mass_matches_all_queries(query_spectra)
+
+        dict_with_preselected_spectra = {}
+        for spectrum_id in same_masses:
+            preselected_matches = same_masses[spectrum_id]
+            dict_with_preselected_spectra[spectrum_id] = \
+                pd.DataFrame(preselected_matches,
+                             columns=["spectrum"])
+        self.collect_data_for_tanimoto_prediction_model(dict_with_preselected_spectra)
+        return dict_with_preselected_spectra
+
+    def get_ms2deepscore_similarity_matrix(
+            self,
+            query_spectra) -> pd.DataFrame:
+        """Returns a dataframe with the ms2deepscore similarity scores
+
+        query_spectra:
+            All query spectra that should get a similarity score with all
+            library spectra.
+        library_embeddings_file_name:
+            File name of the pickled file in which the library embeddings are
+            stored.
+        """
+        ms2ds = MS2DeepScore(self.ms2ds_model)
+        query_embeddings = ms2ds.calculate_vectors(query_spectra)
+        ms2ds_embeddings_numpy = self.ms2ds_embeddings.to_numpy()
+        similarity_matrix = cosine_similarity_matrix(ms2ds_embeddings_numpy,
+                                                     query_embeddings)
+        similarity_matrix_dataframe = pd.DataFrame(
+            similarity_matrix,
+            index=self.ms2ds_embeddings.index,
+            columns=[spectrum.get("spectrum_id") for
+                     spectrum in query_spectra])
+        return similarity_matrix_dataframe
+
+    def _get_spec2vec_similarity_matrix(self,
+                                        query_spectra: List[Spectrum]
+                                        ) -> pd.DataFrame:
+        """
+        Returns a matrix with s2v similarity scores for all query spectra
```

Comment:
```suggestion
        Returns a table (DataFrame) with s2v similarity scores between all query spectra
```

Reply:
Changed to: Returns s2v similarity scores between all query and library spectra
(so it still fits on one line)